src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -41,7 +42,7 @@ public sealed class SonarQubeIssuesProvider : ISonarQubeIssuesProvider, IDisposa
         private readonly ILogger logger;
         private readonly CancellationTokenSource initialFetchCancellationTokenSource;
 
-        private IList<SonarQubeIssue> cachedSuppressedIssues;
+        private List<IGrouping<string, SonarQubeIssue>> cachedSuppressedIssues;
```

Comment:
I haven't seen any performance issue with this data-structure but it might be worth doing some manual test on a big project like Roslyn.